OneSignalSDK/onesignal/core/src/main/java/com/onesignal/internal/OneSignalImp.kt

```diff
@@ -519,6 +512,35 @@ internal class OneSignalImp : IOneSignal, IServiceProvider {
         }
     }
 
+    private fun addFetchParamsObserverAndLoginUser(
+        suppressBackendOperation: Boolean,
+        externalId: String,
+    ) {
+        createAndSwitchToNewUser(suppressBackendOperation = suppressBackendOperation)
+        paramsBackendService!!.addParamsBackendServiceObserver(
+            object : IParamsBackendServiceObserver {
+                override fun onParamsFetched(params: ParamsObject) {
+                    if (params.useIdentityVerification == true) {
+                        // remove identification required operations in the queue
+                        operationRepo!!.removeIdentificationRequiredOperations()
```

Comment:
I don't think this is the right high level concept here. We want to remove any operations called before `OneSignal.login()`. If `addTags()` are call right after login we want to keep it, but if it was call before we want to discard.
* An edge case that comes to mind which is tricky is subscription updates, such as the user accepting the notification permission.